tests/performance/utils.js

```diff
@@ -21,13 +21,67 @@ function median( array ) {
  * @return {string} Result file name.
  */
 function getResultsFilename( fileName ) {
-	const prefixArg = process.argv.find( ( arg ) => arg.startsWith( '--prefix' ) );
-	const fileNamePrefix = prefixArg ? `${prefixArg.split( '=' )[1]}-` : '';
+	const prefixArg = process.argv.find( ( arg ) =>
+		arg.startsWith( '--prefix' )
+	);
+	const fileNamePrefix = prefixArg ? `${ prefixArg.split( '=' )[ 1 ] }-` : '';
 	const resultsFilename = fileNamePrefix + fileName + '.results.json';
 	return resultsFilename;
 }
 
+/**
+ * Returns time to first byte (TTFB) using the Navigation Timing API.
+ *
+ * @see https://web.dev/ttfb/#measure-ttfb-in-javascript
+ *
+ * @return {Promise<number>}
+ */
+async function getTimeToFirstByte() {
+	return page.evaluate(
+		() =>
+			new Promise( ( resolve ) => {
+				new PerformanceObserver( ( entryList ) => {
+					const [ pageNav ] =
+						entryList.getEntriesByType( 'navigation' );
+
+					resolve( pageNav.responseStart - pageNav.startTime );
+				} ).observe( {
+					type: 'navigation',
+					buffered: true,
+				} );
+			} )
+	);
+}
+
+/**
+ * Returns the Largest Contentful Paint (LCP) value using the dedicated API.
+ *
+ * @see https://w3c.github.io/largest-contentful-paint/
+ * @see https://web.dev/lcp/#measure-lcp-in-javascript
+ *
+ * @return {Promise<number>}
+ */
+async function getLargestContentfulPaint() {
+	return page.evaluate(
+		() =>
+			new Promise( ( resolve ) => {
+				new PerformanceObserver( ( entryList ) => {
```

Comment:
Would it be beneficial to free the event's callback memory and closing it when it is no longer useful? In Gutenberg, we use as a [trigger](https://github.com/WordPress/gutenberg/blob/034f49c138ed82c75109df3a37af66c41913b18b/packages/e2e-tests/specs/performance/front-end-block-theme.test.js#L47) the `visibility` change event (when it transitions to `hidden`).

Reply:
Hmm, given that there will be a new navigation right afterwards, I don't see how this would be needed.

I see that the code in Gutenberg was copied from one of Addy's posts/Gists, showing how LCP could be measured over the course of the page's lifecycle. Not really applicable here I'd say.